compile/interface/src/main/scala/xsbt/Dependency.scala

```diff
@@ -63,7 +63,7 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
               case Some((f, className, inOutDir)) =>
                 if (inOutDir && on.isJavaDefined) registerTopLevelSym(on)
                 f match {
-                  case ze: ZipArchive#Entry => for (zip <- ze.underlyingSource; zipFile <- Option(zip.file)) binaryDependency(zipFile, className)
+                  case ze: ZipArchive#Entry => for (zip <- ze.underlyingSource; zipFile <- Option(zip.file).filterNot(_.isDirectory)) binaryDependency(zipFile, className)
```

Comment:
I think this is worth a comment in the source.

Reply:
Something like:

```scala
// This skips the Zip entry whose underlying source is directory to make 2.10 compilation consistent with 2.12
// See https://github.com/sbt/sbt/issues/3142#issuecomment-340947086 
```

?

Reply:
I've pushed a comment.